keps/sig-cloud-provider/20190125-out-of-tree-ibm.md

```diff
@@ -40,60 +43,135 @@ status: provisional
 
+## Terms
+
+- **CCM:** [Cloud Controller Manager](https://kubernetes.io/docs/reference/command-line-tools-reference/cloud-controller-manager/) - The controller manager responsible for
+  running cloud provider dependent logic, such as the service and route
+  controllers.
+- **KCM:** [Kubernetes Controller Manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/) - The controller manager responsible
+  for running generic Kubernetes logic, such as job and node_lifecycle controllers.
+- **IKS:** [IBM Cloud Kubernetes Service](https://www.ibm.com/cloud/container-service)
+- **IBM CCM:** IBM cloud provider using CCM out-of-tree cloud provider architecture.
+- **IBM KCM:** IBM cloud provider using KCM in-tree cloud provider architecture.
+  This cloud provider is currently used by IKS.
+
 ## Summary
 
-Build support for the out-of-tree IBM cloud provider. This involves a well-tested version of the cloud-controller-manager 
-that has feature parity to the kube-controller-manager. 
+Build support for the out-of-tree IBM cloud provider based on the current
+IBM cloud provider used by IBM's managed Kubernetes service (IKS). This involves
+a well-tested version of IBM CCM that has feature parity to the current
+IBM KCM.
+
+Unlike many other cloud providers, IBM's cloud provider was never in-tree due
+to the deprecation of in-tree providers at the time IBM's cloud provider was
+written.
 
 ## Motivation
 
-Motivation for supporting out-of-tree providers can be found in [KEP-0002](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/0002-cloud-controller-manager.md). 
+Motivation for supporting out-of-tree providers can be found in
+[KEP-20180530-cloud-controller-manager](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20180530-cloud-controller-manager.md).
 This KEP is specifically tracking progress for the IBM cloud provider.
 
 ### Goals
 
-* Develop/test/release the IBM cloud-controller-manager
-* Kubernetes clusters running on IBM should be running the cloud-controller-manager.
+* Develop, test, document and release IBM CCM.
+* Kubernetes clusters running on IBM Cloud should be running IBM CCM.
+* IKS clusters should be running IBM CCM.
+* Clusters should be able to easily migrate from IBM KCM to IBM CCM.
 
 ### Non-Goals
 
-* Removing in-tree IBM cloud provider code, this effort falls under the [KEP for removing in-tree providers](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/2019-01-25-removing-in-tree-providers.md).
+* Re-architecting the IBM cloud provider beyond transition from IBM KCM to IBM CCM.
+* Open source other components of IKS.
 
 ## Proposal
 
-### Implementation Details/Notes/Constraints [optional]
-
-TODO for SIG-IBM
+### Implementation Details/Notes/Constraints
+
+IBM has already completed most of the IBM CCM enablement work. Furthermore, IBM
```

Comment:
@rtheis is the plan to just open source the IBM cloud provider or to also host it in the Kubernetes org? Asking since we would only need a tracking KEP _if_ we are hosting it in the k8s org (sorry if this wasn't clear before). If the intent is to host it under the k8s org, we would also need to outline the requirements to qualify for this (see https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/providers/0004-cloud-provider-template.md).

Reply:
@andrewsykim The plan is to host it in the k8s org.

Reply:
I spoke with @andrewsykim and we've decided that this is not the appropriate KEP for IBM's cloud provider.  I'm going to close this and move applicable portions to a new KEP based on  https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/providers/0004-cloud-provider-template.md.